numpy/linalg/lapack_lite/f2c.h

```diff
@@ -8,15 +8,17 @@
 #define F2C_INCLUDE
 
 #include <math.h>
+#include "numpy/npy_common.h"
+#include "npy_cblas.h"
 
-typedef int integer;
+typedef CBLAS_INT integer;
 typedef char *address;
 typedef short int shortint;
 typedef float real;
 typedef double doublereal;
 typedef struct { real r, i; } complex;
 typedef struct { doublereal r, i; } doublecomplex;
-typedef int logical;
+typedef CBLAS_INT logical;
```

Comment:
I'm a little surprised this is the case, but I assume it comes down to the definitions of functions from cblas?

Reply:
`npy_cblas.h` does not contain functions that have `logical` arguments, and numpy.linalg does not use such functions either. (LAPACK does however contain such functions.)

I think there is no standard for size of `logical` in Fortran. For gfortran, `-fdefault-integer-8` implies also LOGICAL is 8 bytes, and I think that's also true for ifort.

The "canonical" C API LAPACKE defines `lapack_logical` as `lapack_int`, also in its ILP64 configuration. https://github.com/Reference-LAPACK/lapack/blob/f24797e40e44c6d98ea7fa4e36bd1ec6c02c95e5/LAPACKE/include/lapacke_config.h#L46-L56